### ISSUE TITLE
Is dimension changed from 1 to 512 when setting features='MS'?

### ISSUE THREAD

**Issue by QuantHao** [zhouhaoyi/Informer2020#95]
Hi, thanks for this thoughtful paper and sharing code base.

When using multi-variate ts to predict uni-variate ts, from
https://github.com/zhouhaoyi/Informer2020/blob/99b59c181f871a1521a798a37287a27ba5b742a7/exp/exp_informer.py#L112-L114
it seems that shape of batch_y is [B, L, 1]. Then it will be project into d_model (default setting 512) as https://github.com/zhouhaoyi/Informer2020/blob/99b59c181f871a1521a798a37287a27ba5b742a7/models/model.py#L23-L24

Should hyperparameter d_model should be set as a much smaller value? Projectting channel 1 to channel 512 really makes me concerned. I don't have a clear answer, and want to discuss more about this issue.

**Comment by cookieminions** [comment]
In MS mode, the input dimension of Decoder will be the same as input dimension of Encoder, and the output dimension of Decoder will be 1.
See code https://github.com/zhouhaoyi/Informer2020/blob/99b59c181f871a1521a798a37287a27ba5b742a7/exp/exp_informer.py#L198-L200
We select the target column of batch_y to calculate loss after we feed the complete batch_y into the Decoder.

**Comment by QuantHao** [comment]
Yes, I understand and agree that output dimension of Decoder is 1, BUT I get confused about your words "input dimension of Decoder will be the same as input dimension of Encoder". From the code https://github.com/zhouhaoyi/Informer2020/blob/410535ebeb59e526a7402e0454235fe1cab83605/exp/exp_informer.py#L119-L121
last axis (channel) of dec_inp is the same as last axis of batch_y. Shouldn't batch_y channel == 1 in MS mode?

**Comment by cookieminions** [comment]
https://github.com/zhouhaoyi/Informer2020/blob/410535ebeb59e526a7402e0454235fe1cab83605/data/data_loader.py#L55-L59

`batch_y` has all features when mode is `MS`, and we set `c_out` of model to 1 and select the target columns of batch_y to calculate the loss.

https://github.com/zhouhaoyi/Informer2020/blob/99b59c181f871a1521a798a37287a27ba5b742a7/exp/exp_informer.py#L198-L200

**Comment by QuantHao** [comment]
That's interesting. If the task is a multi-variate autoregression, I would totally agree with you that channel of batch_y == channel of bath_X because we try to use X(t-1, 1), X(t-1, 2), ...X(t-1, n) to predict X(t, 1), X(t, 2), ...X(t, n). 

However, if people apply your Informer model on a task of exogenous regression which means using X(t-1, 1), X(t-1, 2), ...X(t-1, n) to predict y(t), I cann't get it why decoder input dimension would still be n. It seems to me that 1 might be a more intuitive choice.

**Comment by cookieminions** [comment]
Mode `MS` is not in our paper's experiments, you can adjust it according to your needs.

You can add this code before feed dec_inp into the model to make decoder input's dimension be 1.
```
f_dim = -1 if self.args.features=='MS' else 0
dec_inp = dec_inp[:, :, f_dim:].to(self.device)
```

**Comment by QuantHao** [comment]
I see. Thanks for your thoughts.

**Comment by omerlux** [comment]
> I see. Thanks for your thoughts.

Hi @QuantHao ,
I wanted to know what did you change in the code in order to train the model on multivariate input to predict univariate target.
Thank!
